src/affordance/tasks/arxiv_summarization.py

```diff
@@ -0,0 +1,368 @@
+        auto_cot_prompt += cot[0] + "\n----\n"
+
+    print(auto_cot_prompt)
+    f = open("auto_cot_demonstrations.txt", "a+")
+    f.write("Anachronisms\n\n")
+    f.write(auto_cot_prompt)
+
+    if not predict:
+        return
+
+    def predict(chunk):
+        gpt3 = OpenAIModel(
+            model=model_name, max_length=500, temperature=temperature, quote="---", n=1
+        )
+        prompts = [
+            auto_cot_prompt
+            + """%s\n""" % description
+            + """%s\nA: Let's think step-by-step.\n""" % x
+            for x in chunk
+        ]
+        return gpt3(prompts)
+
+    predictions_list = []
+    labels_list = []
+    runs = 5
+    for run in range(runs):
+        print("Run %d" % run)
+        answers = []
+        for x in tqdm(
+            chunks(inputs, 10)
+        ):  # should we replace the chunks function by inputs.split('\n') to use paragraphs ? or use the chunks function with longer chunks?
+            x = [ex.replace("\nA:", "") for ex in x]
+            answers.extend(predict(x))
+            # time.sleep(10)
+        preds = [get_autocot_answer(x) for x in answers]
+        predictions_list.append(preds)
+        labels_list.append(labels)
+    print("Rouge performance")
+    print(rouge.compute(predictions=predictions_list, references=labels_list))
+
+
+def nl_program(
+    temperature=0.3,
+    model_name="davinci-codex-002-msft",
+    strategy="fixed",
+    self_consistency=False,
+):
+    global few_shot_cot_prompt
+    task_name = "Article Summarization"
+    task_description = "(Physics Questions) Answer these high-school-level physics questions by applying the right physics formula, making substitutions, and storing the result in 'ans'."
+
+    if strategy == "fixed":
+        few_shot_cot_prompt = few_shot_cot_prompt
+    elif strategy == "random":
+        few_shot_cot_prompt = random_tasks(N=6)
+    elif strategy == "similar":
+        few_shot_cot_prompt = similar_tasks(task_description, io_pairs, N=6)
+    elif strategy == "similar_auto_decomp":
+        few_shot_cot_prompt = similar_auto_breakdowns(task_description, io_pairs, N=6)
+    elif strategy == "llm_similar":
+        few_shot_cot_prompt = llm_similar_tasks(task_name, task_description, io_pairs, N=6)
+
+    interpreter = TopDownVisitor(model_name=model_name, exclude_list=["[generate python code]"])
+
+    def predict(description, chunk):
+        gpt3 = OpenAIModel(
+            model=model_name, max_length=1000, temperature=temperature, quote="---", n=1
+        )
+        prompts = [few_shot_cot_prompt % (description, x) for x in chunk]
+        return prompts, gpt3(prompts)
+
+    def predict_self_consistency(description, chunk, n=15):
+        gpt3 = OpenAIModel(
+            model=model_name, max_length=1000, temperature=temperature, quote="---", n=n
+        )
+        prompts = [few_shot_cot_prompt % (description, x) for x in chunk]
+        return prompts, gpt3(prompts)
+
+    if self_consistency:
```

Comment:
This still looks like the old implementation before I refactored it; mind updating with the newest stuff from physics_questions.py?